### PR TITLE
remove assert and code cleanup

### LINKED ISSUE #1533: Don't use `assert` outside of tests
### Description of feature

The problem with assert is that it is not executed if python is invoked with optimizations. This means that the code behaves differently and this is probably not desired.

Use `git grep assert nf_core` to find where the problem lies.

See [this stackoverflow answer](https://stackoverflow.com/a/5142453/6018688) for explanations and also [this biopython issues](https://github.com/biopython/biopython/issues/3600) for an example of a codebase that followed the suggestion and more info.

### PR DESCRIPTION
**Pull request by fabianegli**
This PR removes all assert statements from production code.

It also contains minuscule code cleanup and speedups.

Closes #1533 

## PR checklist

- [x] This comment contains a description of changes (with reason)
- [x] `CHANGELOG.md` is updated
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] Documentation in `docs` is updated

### CODE REVIEW

File: nf_core/lint/nextflow_config.py [hunk 1]
```diff
@@ -206,24 +206,24 @@ def nextflow_config(self):
...
+            manifest_name = self.nf_config.get("manifest.name", "").strip("'\"")
+            if not manifest_name.startswith("nf-core/"):
+                raise AssertionError()
         except (AssertionError, IndexError):
-            failed.append(
-                "Config ``manifest.name`` did not begin with ``nf-core/``:\n    {}".format(
-                    self.nf_config.get("manifest.name", "").strip("'\"")
-                )
-            )
+            failed.append("Config ``manifest.name`` did not begin with ``nf-core/``:\n    {}".format(manifest_name))
```
Comment (ErikDanielsson) [inline nf_core/lint/nextflow_config.py:213]:
```suggestion
            failed.append(f"Config ``manifest.name`` did not begin with ``nf-core/``:\n    {manifest_name}")
```

**Review by ErikDanielsson — COMMENTED** [review]
Looks good to me! The changelog needs updating before we merge though.

File: CHANGELOG.md [hunk 2]
```diff
@@ -23,6 +23,7 @@
 
 - Remove support for Python 3.6 ([#1680](https://github.com/nf-core/tools/pull/1680))
 - Add support for Python 3.9 and 3.10 ([#1680](https://github.com/nf-core/tools/pull/1680))
+- Programm control flow no longer affected when run with optimizations ([#1685](https://github.com/nf-core/tools/pull/1685))
```
Comment (ErikDanielsson) [inline CHANGELOG.md:26]:
```suggestion
- Program control flow no longer affected when run with optimizations ([#1685](https://github.com/nf-core/tools/pull/1685))
```

**Review by ErikDanielsson — APPROVED** [review]
Looks good, just a typo.

File: CHANGELOG.md (hunk 2 above)
Comment (fabianegli) [reply CHANGELOG.md:26]:
```suggestion
- Invoking Python with optimizations no longer affects the program control flow ([#1685](https://github.com/nf-core/tools/pull/1685))
```

**Pull request merged**